lib/posix/utsname.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <zephyr/posix/sys/utsname.h>
+
+int uname(struct utsname *name)
+{
+	if (name == NULL) {
+		return -1;
+		errno = EINVAL;
```

Comment:
https://pubs.opengroup.org/onlinepubs/9699919799/functions/uname.html:

> Upon successful completion, a non-negative value shall be returned. Otherwise, -1 shall be returned and errno set to indicate the error.

Reply:
I wouldn't even bother with the runtime check.  "No errors defined".
https://pubs.opengroup.org/onlinepubs/9699919799/functions/uname.html

Reply:
Replaced that runtime check with `__ASSERT_NO_MSG`